src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -55,6 +58,9 @@ class ControllerScriptEngineLegacyTest : public ControllerScriptEngineLegacy, pu
 
     void TearDown() override {
         mixxx::Time::setTestMode(false);
+#ifdef MIXXX_USE_QML
+        m_rootItems.clear();
+#endif
```

Comment:
does it work without the explicit `.clear()`? Ideally RAII would take care of this fully by itself.

Reply:
It will probably work, but I preferred to make it explicit so if something was to crash when deallocating the screen, it still happens as part of the test hooks

Reply:
makes sense. It was more of a general question whether this would be needed in production code.